app/src/main/java/com/tananaev/passportreader/MainActivity.kt

```diff
@@ -164,11 +165,22 @@ abstract class MainActivity : AppCompatActivity() {
         super.onResume()
         val adapter = NfcAdapter.getDefaultAdapter(this)
         if (adapter != null) {
-            val intent = Intent(applicationContext, this.javaClass)
-            intent.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP
-            val pendingIntent = PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_MUTABLE)
-            val filter = arrayOf(arrayOf("android.nfc.tech.IsoDep"))
-            adapter.enableForegroundDispatch(this, pendingIntent, null, filter)
+            if (adapter.isEnabled) {
+                val intent = Intent(applicationContext, this.javaClass)
+                intent.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP
+                val pendingIntent =
+                    PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_MUTABLE)
+                val filter = arrayOf(arrayOf("android.nfc.tech.IsoDep"))
+                adapter.enableForegroundDispatch(this, pendingIntent, null, filter)
+            } else {
+                Toast.makeText(
+                    applicationContext,
+                    getString(R.string.enable_nfc_toast),
+                    Toast.LENGTH_LONG
+                ).show();
+                Thread.sleep(5_000)
```

Comment:
Why do we need sleep?

Reply:
Otherwise the user will not see the toast message because the ACTION_NFC_SETTINGS activity intent will open a new window that does some explaining for the user about why one needs to enable NFC.

Reply:
I think we should do a proper dialog instead of this.

Reply:
I am not sure what you have in mind, but I will close this PR then. :)